docs/src/main/asciidoc/elasticsearch.adoc

```diff
@@ -66,21 +65,21 @@ For the Elasticsearch low level client, add:
 implementation("io.quarkus:quarkus-elasticsearch-rest-client")
 ----
 
-For the Elasticsearch high level client, add:
+For the Elasticsearch Java API client, add:
```

Comment:
Unrelated to your PR, but this got me thinking that we should probably split the clients some time in the future. The guide doesn't quite flow naturally with all these "Or if you want this instead, do that".